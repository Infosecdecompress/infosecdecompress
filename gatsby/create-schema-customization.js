const createSchemaCustomization = ({ actions }) => {
  const { createTypes } = actions;
  const typeDefs = `
    type MarkdownRemark implements Node {
      frontmatter: Frontmatter
    }
    type SocialImage {
      publicURL: String
    }
    type Frontmatter {
      socialImage: SocialImage
    }
  `;

  createTypes(typeDefs);
};

module.exports = createSchemaCustomization;