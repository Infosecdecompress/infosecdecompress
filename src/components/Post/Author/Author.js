// @flow strict
import React from 'react';
import { getContactHref } from '../../../utils';
import styles from './Author.module.scss';
import { useSiteMetadata } from '../../../hooks';

const Author = () => {
  const { author } = useSiteMetadata();

  return (
    <div className={styles['author']}>
      <p className={styles['author__bio']}>
      </p>	  
    </div>
  );
};

export default Author;
