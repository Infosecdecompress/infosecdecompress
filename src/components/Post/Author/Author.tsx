import React from 'react';
import { getContactHref } from '../../../utils';
import * as styles from './Author.module.scss';
import { useSiteMetadata } from '../../../hooks';

const Author = () => {
  const { author } = useSiteMetadata();

  return (
    <div className={styles['author']}>
      <center><p className={styles}>
        資安解壓縮網站內容除特別註記外皆以
        <a 
          href="http://creativecommons.org/licenses/by-sa/4.0/"
          rel="noopener noreferrer"
        >
          CC BY-SA 4.0
        </a>
        方式授權
        <br/>
        對於這個節目有什麼想法，或想要聽什麼內容都歡迎到 
        <a
          href="/pages/contacts"
          rel="noopener noreferrer"
        >
          聯絡資訊
        </a>
        告訴我們
      </p></center>
    </div>
  );
};

export default Author;
