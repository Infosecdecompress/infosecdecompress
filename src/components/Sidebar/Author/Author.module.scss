@import "../../../assets/scss/variables";
@import "../../../assets/scss/mixins";

.author {
  .photo {
    background-clip: padding-box;
    border-radius: 50%;
    display: inline-block;
    height: 75px;
    width: 75px;
    @include margin-bottom(0);
  }

  .title {
    font-size: $typographic-base-font-size * 1.125;
    font-weight: 600;
    @include line-height(1.125);
    @include margin(0.5, 0, 0.5, 0);

    .link {
      color: $color-base;

      &:hover,
      &:focus {
        color: $color-base;
      }
    }
  }

  .subtitle {
    color: $color-gray;
    @include line-height(1);
    @include margin-bottom(1);
  }
}
