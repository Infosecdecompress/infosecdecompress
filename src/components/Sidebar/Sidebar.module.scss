@import "../../assets/scss/variables";
@import "../../assets/scss/mixins";

.sidebar {
  width: 100%;

  .inner {
    @include padding(1, 0.75, 0);
    position: relative;
  }
}

@include breakpoint-sm {
  .sidebar {
    lost-column: 5/12;

    .inner {
      @include padding(1.25, 0.75, 0);

      &:after {
        background: $color-gray-border;
        background: linear-gradient(
          to bottom,
          $color-gray-border 0%,
          $color-gray-border 48%,
          $color-white 100%
        );
        bottom: 0;
        content: "";
        height: 540px;
        position: absolute;
        right: -10px;
        top: 30px;
        width: 1px;
      }
    }
  }
}

@include breakpoint-md {
  .sidebar {
    lost-column: 1/3;

    .inner {
      @include padding-equal(1.5);
    }
  }
}
