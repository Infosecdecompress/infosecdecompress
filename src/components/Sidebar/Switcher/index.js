// @flow strict
export { default } from './Switcher';
