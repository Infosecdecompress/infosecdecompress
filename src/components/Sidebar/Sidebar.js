// @flow strict
import React from 'react';
import Author from './Author';
import Contacts from './Contacts';
import Copyright from './Copyright';
import Menu from './Menu';
import styles from './Sidebar.module.scss';
import { useSiteMetadata } from '../../hooks';
import Search from "../SearchContainer";
import Switcher from './Switcher';

const siteConfig = require('../../../config');

type Props = {
  isIndex?: boolean,
};

const Sidebar = ({ isIndex }: Props) => {
  const { author, copyright, menu } = useSiteMetadata();
  const { darkFunctionalityIsOn } = siteConfig;

  return (
    <div className={styles['sidebar']}>
      <div className={styles['sidebar__inner']}>
        <Author author={author} isIndex={isIndex} />
        <Menu menu={menu} />
        <Contacts contacts={author.contacts} />
        <Search />
        {darkFunctionalityIsOn ? <Switcher /> : null}
        <Copyright copyright={copyright} />
        </div>
      </div>
  );
};

export default Sidebar;
