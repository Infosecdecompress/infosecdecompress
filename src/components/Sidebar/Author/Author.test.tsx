import React from "react";
import renderer from "react-test-renderer";

import { StaticQuery } from "gatsby";

import { Author } from "@/components/Sidebar/Author";
import * as mocks from "@/mocks";

const mockedStaticQuery = StaticQuery as jest.Mock;

describe("Author", () => {
  beforeEach(() => {
    mockedStaticQuery.mockImplementationOnce(() => null);
  });

  it("renders correctly", () => {
    const props = { isIndex: false, author: mocks.author };
    const tree = renderer.create(<Author {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
