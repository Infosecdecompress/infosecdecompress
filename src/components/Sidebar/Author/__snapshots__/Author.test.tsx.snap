// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Author renders correctly 1`] = `
<div>
  <a
    href="/"
  />
  <h2>
    <a
      href="/"
    >
      John Doe
    </a>
  </h2>
  <p>
    Pellentesque odio nisi, euismod in, pharetra a, ultricies in, diam. Sed arcu.
  </p>
</div>
`;
